Messaging Pattern API Specifications (#151)
eclipse-uprotocol/up-spec: up-l2/README.adoc

SPDX-License-Identifier: Apache-2.0
----

uProtocol Communication layer (uP-L2) defines the message patterns (pub/sub, notification, and RPC) APIs used by applications & services (uEs), as well as the routing and dispatching rules for transmitting said communication messages across distributed heterogeneous systems. 

.uP-L2 Sections
[width="80%",cols="30%,70%",options="header"]
|===
|Section | Description

| link:dispatchers/README.adoc[*Dispatchers*]
| uEs that are responsible for dispatching/forwarding messages between uEs and between uDevices.

| link:api.adoc[*APIs*]

> |  Communication Layer API that implements the pub/sub, notification, and RPC messaging patterns for uE developers (applications and/or services) can use.

Reviewer: ```suggestion
|  Communication Layer API that allows uEntity developers (applications and/or services) to employ the pub/sub, notification and RPC messaging patterns.
```
